subsys/net/lib/lwm2m/lwm2m_obj_portfolio.c

```diff
@@ -0,0 +1,129 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdint.h>
+#include <init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_resource_ids.h"
+
+#define PORTFOLIO_VERSION_MAJOR 1
+#define PORTFOLIO_VERSION_MINOR 0
+
+#define PORTFOLIO_IDENTITY_ID			0
+#define PORTFOLIO_GET_AUTH_DATA_ID		1
+#define PORTFOLIO_AUTH_DATA_ID			2
+#define PORTFOLIO_AUTH_STATUS_ID		3
+
+#define PORTFOLIO_MAX_ID 4
+
+/* Identity max based on LwM2M v1.1 conformance test requiremnts */
+#define PORTFOLIO_IDENTITY_MAX 4
+#define PORTFOLIO_AUTH_DATA_MAX 4
+/* Support only 1 even this multi instance object */
+#define MAX_INSTANCE_COUNT	1
+
+/*
+ * Calculate resource instances as follows:
+ * start with PORTFOLIO_MAX_ID
+ * subtract MULTI resources and execute because their counts include 0 resource (4)
+ * add PORTFOLIO_IDENTITY_ID resource instances
+ * add PORTFOLIO_AUTH_DATA_ID resource instances
+ */
+#define RESOURCE_INSTANCE_COUNT        (PORTFOLIO_MAX_ID - 3 + \
+					PORTFOLIO_IDENTITY_MAX + \
+					PORTFOLIO_AUTH_DATA_MAX)
+
+
+static struct lwm2m_engine_obj portfolio;
+static struct lwm2m_engine_obj_field fields[] = {
+	OBJ_FIELD_DATA(PORTFOLIO_IDENTITY_ID, RW, STRING), /* Mandaory Multi Instance */
+	OBJ_FIELD_EXECUTE(PORTFOLIO_GET_AUTH_DATA_ID), /* Optional */
+	OBJ_FIELD_DATA(PORTFOLIO_AUTH_DATA_ID, R_OPT, OPAQUE), /* Optional Multi instance */
+	OBJ_FIELD_DATA(PORTFOLIO_AUTH_STATUS_ID, R_OPT, U8), /* Optional: Single */
+};
+
+static struct lwm2m_engine_obj_inst inst[MAX_INSTANCE_COUNT];
+static struct lwm2m_engine_res res[MAX_INSTANCE_COUNT][BUTTON_MAX_ID];
+static struct lwm2m_engine_res_inst res_inst[MAX_INSTANCE_COUNT][RESOURCE_INSTANCE_COUNT];
+
+static int portfolio_execute_cb(uint16_t obj_inst_id, uint8_t *args, uint16_t args_len)
+{
+        /* Portfolio object is just LwM2M minimal implemtation so we not handle this */
+	return -EPERM;
+}
+
+static struct lwm2m_engine_obj_inst *portfolio_create(uint16_t obj_inst_id)
+{
+	int index, avail = -1, i = 0, j = 0;
+
+	/* Check that there is no other instance with this ID */
+	for (index = 0; index < ARRAY_SIZE(inst); index++) {
+		if (inst[index].obj && inst[index].obj_inst_id == obj_inst_id) {
+			LOG_ERR("Can not create instance - "
+				"already existing: %u", obj_inst_id);
+			return NULL;
+		}
+
+		/* Save first available slot index */
+		if (avail < 0 && !inst[index].obj) {
+			avail = index;
+		}
+	}
+
+	if (avail < 0) {
+		LOG_ERR("Can not create instance - no more room: %u",
+			obj_inst_id);
+		return NULL;
+	}
+
+	button_data[avail].obj_inst_id = obj_inst_id;
+
+	(void)memset(res[avail], 0,
+		     sizeof(res[avail][0]) * ARRAY_SIZE(res[avail]));
+	init_res_instance(res_inst[avail], ARRAY_SIZE(res_inst[avail]));
+
+	/* initialize instance resource data */
+        INIT_OBJ_RES_MULTI_OPTDATA(PORTFOLIO_IDENTITY_ID, res, i, res_inst, j,
+				   PORTFOLIO_IDENTITY_MAX, false);
+        INIT_OBJ_RES_EXECUTE(PORTFOLIO_GET_AUTH_DATA_ID, res[index], i,
```

Comment:
Maybe just set the handler to NULL if it's a dummy one anyway?

Reply:
OK I will set NULL